core/trino-main/src/main/java/io/trino/server/Server.java

```diff
@@ -167,7 +167,8 @@ private void doStart(String trinoVersion)
             injector.getInstance(optionalKey(HeaderAuthenticatorManager.class))
                     .ifPresent(HeaderAuthenticatorManager::loadHeaderAuthenticator);
 
-            injector.getInstance(optionalKey(OAuth2Client.class)).ifPresent(OAuth2Client::load);
+            Set<LoadableComponent> loadableComponents = injector.getInstance(Key.get(new TypeLiteral<>() {}));
+            loadableComponents.forEach(LoadableComponent::load);
```

Comment:
This is an idea but in the future we could consider replacing the below with `LoadableCompoent`s. I think this is useful especially for optional binder:
<img width="725" alt="image" src="https://user-images.githubusercontent.com/7569403/225923179-fb93857b-22b3-4153-b69f-e240156fbc16.png">